### PR TITLE
feat(growth): adding create sample transaction button

### PR DESCRIPTION
**Pull request by PixelDust22**
Implement the design for button to create sample transactions. For now, only creates JS sample transactions.

image

### CODE REVIEW

**Review by scefali — COMMENTED** [review]
@Neo-Zhixing maybe you could slightly adjust the illustration so the "Take a tour" text isn't overlapping with the line in the illustration

File: static/app/views/performance/onboarding.tsx [hunk 1]
```diff
@@ -101,6 +107,54 @@ function Onboarding({organization}: Props) {
...
+  const secondaryBtn = showSampleTransactionBtn ? (
+    <Button
+      onClick={async () => {
+        trackAdvancedAnalyticsEvent(
+          'growth.performance_sample_transaction',
+          {platform: project.platform},
+          organization
+        );
+        const url = `/projects/${organization.slug}/${project.slug}/create-sample-transaction/`;
+        const eventData = await api.requestPromise(url, {method: 'POST'});
```
Comment (scefali) [inline static/app/views/performance/onboarding.tsx:147]: @Neo-Zhixing we should have some logic to handle errors here

File: static/app/views/performance/onboarding.tsx [hunk 2]
```diff
@@ -101,6 +107,54 @@ function Onboarding({organization}: Props) {
...
+      onAdvance={handleAdvance}
+      onCloseModal={handleClose}
+      doneUrl={performanceSetupUrl}
+      doneText={t('Start Setup')}
+    >
+      {({showModal}) => (
+        <Button
+          priority={showSampleTransactionBtn ? 'link' : 'default'}
+          onClick={() => {
+            trackAnalyticsEvent({
```
Comment (scefali) [inline static/app/views/performance/onboarding.tsx:125]: @Neo-Zhixing mind updating to the `trackAdvancedAnalyticsEvent` function heree? We can make a new file for Performance events (see static/app/utils/issueEvents.tsx as an example).

File: static/app/views/performance/content.tsx [hunk 3]
```diff
@@ -223,8 +225,8 @@ class PerformanceContent extends Component<Props, State> {
           </PageHeader>
           <GlobalSdkUpdateAlert />
           {this.renderError()}
-          {showOnboarding ? (
-            <Onboarding organization={organization} />
+          {showOnboarding && project ? (
```
Comment (scefali) [inline static/app/views/performance/content.tsx:228]: @Neo-Zhixing what happens if `project` is falsy? Is that possible?

File: static/app/views/performance/onboarding.tsx [hunk 4]
```diff
@@ -101,6 +107,54 @@ function Onboarding({organization}: Props) {
...
+          {t('Take a Tour')}
+        </Button>
+      )}
+    </FeatureTourModal>
+  );
+  const secondaryBtn = showSampleTransactionBtn ? (
+    <Button
+      onClick={async () => {
+        trackAdvancedAnalyticsEvent(
+          'growth.performance_sample_transaction',
```
Comment (scefali) [inline static/app/views/performance/onboarding.tsx:142]: @Neo-Zhixing let's make this a `performance_views` event to fit in with the other events on this page.

File: static/app/views/performance/content.tsx (hunk 3 above)
Comment (Neo-Zhixing) [reply static/app/views/performance/content.tsx:228]: `project` can't be falsy. See the `shouldShowOnboarding` function. This is mostly there to make TypeScript happy.

**Review by scefali — APPROVED** [review]

**Pull request merged**